druid/src/win_handler.rs

```diff
@@ -306,14 +306,41 @@ impl<T: Data + 'static> AppState<T> {
 
     fn connect(&mut self, id: WindowId, handle: WindowHandle) {
         self.windows.connect(id, handle);
+        if let Some(delegate) = &mut self.delegate {
```

Comment:
So these 10 lines or so are the main place here where I wonder if we can't reduce boilerplate. What I am imagining is a function on `self` with the signature,

```rust
fn with_delegate(&mut self, f: impl Fn(&mut AppDelegate, &mut DelegateCtx));
```

which internally does this whole dance of destructing self and creating the `DelegateCtx`, and then calls the passed closure with them.

If we had this, we could replace these 12 lines with,

```rust
self.with_delegate(|del, ctx| del.window_added(id, data, env, ctx));
```

Does that make sense?